src/PreCICE.jl

```diff
@@ -1332,5 +1340,159 @@ function actionReadIterationCheckpoint()
     return unsafe_string(msgCstring)
 end
 
+@doc """
+
+    isGradientDataRequired(dataID::Integer)::Bool
+        
+Checks if the given data set requires gradient data. We check if the data object has been intialized with the gradient flag.
+# Arguments
+- `dataID::Integer`: ID of the data to be checked. Obtained by [`getDataID`](@ref).
+
+"""
+function isGradientDataRequired(dataID::Integer)
+        return ccall((:precicec_isGradientDataRequired, "libprecice"), Cint, (Cint,), dataID)
+end
+
+@doc """
+
+    writeBlockVectorGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+
+
+Write gradient data of a vector data as a block, the value of a specified vertices to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndices::AbstractArray{Cint}`: Indices of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 2D problem with 2 vertices:
+
+```julia
+data_id = 1
+valueIndices = [1,2,3]
+gradientValues = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0]
+writeBlockVectorGradientData(data_id, valueIndices, gradientValue)
+```
+
+"""
+function writeBlockVectorGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+    _size, dimensions = size(gradientValues)
+    @assert dimensions == getDimensions()*getDimensions() "Dimensions of vector data in write_block_vector_gradient_data does not match with dimensions in problem definition. Provided dimensions: $dimensions, expected dimensions: $(getDimensions()*getDimensions())"
+
+    ccall((:precicec_writeBlockVectorGradientData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cint}, Ref{Cdouble}), dataID, _size, valueIndices, gradientValues)
+end
```

Comment:
The Values are given as a Matrix but have to be passed as a flattened c vector. Make sure the flattening yields the correct order, see eg `writeBlockVectorData`

Reply:
Fixed. Thanks